File: 0075-sort-colors/0075-sort-colors.py
class Solution(object):
    def sortColors(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        cntz = 0
        cnto = 0
        cntt = 0

        for i in range(len(nums)):
            if nums[i] == 0:
                cntz+=1
            elif nums[i] == 1:
                cnto+=1
            else:
                cntt+=1

        j = 0
        while(cntz>0):
            nums[j] = 0
            j+=1
            cntz-=1

        while(cnto>0):
            nums[j] = 1
            j+=1
            cnto-=1

        while(cntt>0):
            nums[j] = 2
            j+=1
            cntt-=1

        return nums

File: 0075-sort-colors/0075-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int n = nums.size();
        int zc = 0, oc = 0, tc = 0;
        for(int num:nums){
            if(num == 0) zc++;
            else if(num == 1) oc++;
            else tc++;
        }

        for(int i=0;i<n;i++){
            if(zc>0){
                nums[i] = 0; zc--;
            }
            else if(oc>0){
                nums[i] = 1; oc--;
            }
            else{
                nums[i] = 2; tc--;
            }
        }

        // int i=0;
        // while(zc>0){
        //     nums[i] = 0;
        //     i++;
        // }
        // while(oc>0){
        //     nums[i] = 1;
        //     i++;
        // }
        // while(tc>0){
        //     nums[i] = 2;
        //     i++;
        // }
    }
};